### PR TITLE
Basic bun integration

### PR DESCRIPTION
**Pull request by deemkeen**
**Summary**

Basic bun integration as requested in #1106
Bun still doesn't work with windows, so i've tried to skip the windows integration tests, maybe there is a better way to do it

### CODE REVIEW

**Comment by eirslett** [comment]
I'm a bit worried about scope creep for the frontend-maven-plugin. Would it make sense to make something separate, like bun-maven-plugin?

**Comment by deemkeen** [comment]
> I'm a bit worried about scope creep for the frontend-maven-plugin. Would it make sense to make something separate, like bun-maven-plugin?

Maybe it could be an option, on the other hand this plugin is quite universal, it also supports yarn, so i've thought it would be a good idea to add this feature also

File: frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/BunInstaller.java [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        } catch (ProcessExecutionException e) {
+            this.logger.warn("Unable to determine current bun version: {}", e.getMessage());
+            return false;
+        }
+    }
+
+    private void installBunDefault() throws InstallationException {
+        try {
+
+            logger.info("Installing Yarn version {}", bunVersion);
```
Comment (tfactor2) [inline frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/BunInstaller.java:100]: typo in log message: Yarn -> Bun

File: frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/BunInstaller.java [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+            CacheDescriptor cacheDescriptor = new CacheDescriptor("bun", this.bunVersion,
+                    "zip");
+
+            File archive = this.config.getCacheResolver().resolve(cacheDescriptor);
+
+            downloadFileIfMissing(downloadUrl, archive, this.userName, this.password);
+
+            File installDirectory = getInstallDirectory();
+
+            // We need to delete the existing yarn directory first so we clean out any old files, and
```
Comment (tfactor2) [inline frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/BunInstaller.java:114]: Same copy/paste typo problem as in some other places, couldn't the code be shared?

File: frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/BunInstaller.java (hunk 1 above)
Comment (deemkeen) [reply frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/BunInstaller.java]: thanks, fixed

File: frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/BunInstaller.java (hunk 2 above)
Comment (deemkeen) [reply frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/BunInstaller.java]: fix with previous commit, thank you

**Comment by deemkeen** [comment]
@eirslett So as i understand, it's an organizational problem to integrate this because it's difficult to maintain so many technologies? Or is there something else to consider? Should this PR be closed?

**Comment by eirslett** [comment]
Yes it's mostly organizational. More features keep getting requested and added, but there's limited time for maintenance, bug fixes etc. But I triggered the test suite now. I suppose we can merge the bun stuff, it's a promising runtime!

**Comment by eirslett** [comment]
Looks like the MacOS build is failing on `bun install` for some reason...

**Comment by eirslett** [comment]
It was a hiccup!

**Comment by eirslett** [comment]
@deemkeen could you please have a look at updating support here, so Bun also can be used on Windows?

**Comment by deemkeen** [comment]
@eirslett yeah sure, I'll take a look at it this weekend!

**Comment by deemkeen** [comment]
I'm getting strange crashes, while trying to run the tests with the windows runner. Does anyone have any ideas?

`
[INFO] [DEBUG] Executing command line [D:\a\frontend-maven-plugin\frontend-maven-plugin\frontend-maven-plugin\target\it\bun-integration\target\bun, install]
[INFO] [INFO] 
[INFO] [INFO] Panic: start index is larger than end index
[INFO] [INFO] 
[INFO] [INFO] bun has crashed :'(
[INFO] [INFO] 
[INFO] [INFO] ----- bun meta -----
[INFO] [INFO] Bun v1.1.2 (c8d072c2) Windows x64 
[INFO] [INFO] Unknown: 
[INFO] [INFO] Elapsed: 0ms | User: 0ms | Sys: 0ms
[INFO] [INFO] RSS: 16.79MB | Peak: 16.79MB | Commit: 62.39MB | Faults: 4255
[INFO] [INFO] ----- bun meta -----
[INFO] [INFO] 
[INFO] [INFO] 0   00007FF74836E048
[INFO] [INFO] 1   ???
[INFO] [INFO] 2   ???
[INFO] [INFO] 3   ???
[INFO] [INFO] 4   ???
[INFO] [INFO] 5   ???
[INFO] [INFO] 6   ???
[INFO] [INFO] 
[INFO] [INFO] Search GitHub issues https://bun.sh/issues or join in #windows channel in https://bun.sh/discord
[INFO] [INFO] 
[INFO] [INFO] thread 3632 panic: start index is larger than end index
[INFO] [INFO] ???:?:?: 0x7ff746fae7de in ??? (bun)
[INFO] [INFO] ???:?:?: 0x7ff746cd1603 in ??? (bun)
[INFO] [INFO] ???:?:?: 0x7ff74986e033 in ??? (bun)
[INFO] [INFO] ???:?:?: 0x7ffe38334caf in ??? (KERNEL32.DLL)
[INFO] [INFO] ???:?:?: 0x7ffe3863e8aa in ??? (ntdll.dll)
[INFO] [INFO] ------------------------------------------------------------------------
[INFO] [INFO] BUILD FAILURE
`

**Comment by eirslett** [comment]
@deemkeen Not sure! Maybe they have ironed out the issues by now? @giverly-devops if you're able to, you could take a stab at it! (See #1146)

**Pull request merged**